src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+                for (String uniprotResponseLine : uniprotResponseLines) {
+                    List<String> tabSplit = Arrays.asList(uniprotResponseLine.split("\t"));
+                    if (tabSplit.size() == 4) {
+                        uniprotIdLines.add(uniprotResponseLine + "\n");
+                    }
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        // Write data to file.
+        if (!uniprotIdLines.isEmpty()) {
+            File speciesUniprotFile = createNewFile(this.destination + biomartSpeciesName + "_uniprot");
+            for (String uniprotIdLine : uniprotIdLines) {
+                Files.write(Paths.get(speciesUniprotFile.toString()), uniprotIdLine.getBytes(), StandardOpenOption.APPEND);
+            }
+        }
+    }
+
+    private String getMicroarrayTypesQuery(String biomartSpeciesName) {
+        return this.getDataURL() + microarrayTypesBaseQuery.replace("BIOMART_SPECIES_NAME", biomartSpeciesName);
+    }
+
+    /**
+     * Query Biomart for Microarray 'identifiers' associated with the microarray type
+     * @param biomartSpeciesName -- String of species name in Biomart format (eg: hsapiens)
+     * @param microarrayTypes -- String, type of microarray identifiers being queried for.
+     * @throws Exception
+     */
+    private void retrieveAndStoreMicroarrayData(String biomartSpeciesName, String biomartXMLPath, Set<String> microarrayTypes) {
+
+        // Query Biomart for microarray identifiers associated with microarray type.
+        for (String microarrayType : microarrayTypes) {
+            try {
+                logger.info("Retrieving data associated with microarray type " + microarrayType);
+                Set<String> microarrayResponseLines = queryBiomart(getBiomartIdentifierQuery(biomartXMLPath, biomartSpeciesName, microarrayType), 0);
+                Set<String> microarrayIdentifierLines = new HashSet<>();
+                for (String microarrayResponseLine : microarrayResponseLines) {
+                    List<String> tabSplit = Arrays.asList(microarrayResponseLine.split("\t"));
+                    // Microarray ID is 4th column, and sometimes it isn't in a line.
+                    // Since we need the ID, we require the List to have 4 elements.
+                    if (tabSplit.size() == 4) {
+                        microarrayIdentifierLines.add(microarrayResponseLine + "\n");
+                    }
+                }
+
+                // Write data to file.
+                if (!microarrayIdentifierLines.isEmpty()) {
+                    File speciesMicroarrayFile = createNewFile(this.destination + biomartSpeciesName + "_microarray");
+                    for (String probeIdLine : microarrayIdentifierLines) {
+                        Files.write(Paths.get(speciesMicroarrayFile.toString()), probeIdLine.getBytes(), StandardOpenOption.APPEND);
+                    }
```

Comment:
This is similar to the logic for populating the file above.  A generic helper method passing in the name of the file and the set of lines to write would be useful.  You could surround that call with the `if (!lines.isEmpty())` check.